[prometheus] Update prometheus to 2.7.1

predominant opened the pull request:
Signed-off-by: Graham Weldon <graham@grahamweldon.com>

Also: bypass cgo user requirements for build (env user).

### Testing

This was done in a couple of steps, because I was having supervisor problems at the time.

```
hab studio enter
build prometheus
source results/last_build.env
hab pkg install --binlink --force results/${pkg_artifact}
hab svc load ${HAB_ORIGIN}/${pkg_name}
SKIPBUILD=1 ./prometheus/tests/test.sh
```

### Sample output

```
 ✓ Command is on path
 ✓ Version matches
 ✓ Help command
 ✓ Service is running
 ✓ Listening on port 9090

5 tests, 0 failures
```

thesentinels commented:
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

smacfarlane commented on prometheus/plan.sh:
```diff
@@ -2,12 +2,12 @@ pkg_name=prometheus
 pkg_bin_dirs=(bin)
 pkg_source="https://github.com/prometheus/prometheus/archive/v${pkg_version}.tar.gz"
-pkg_shasum=4f84697f746a81df45e434a27b4b6b2c23c08768aa0f68b7bcf88f4ee6d1d8d6
+pkg_shasum=bfbeb434342a03b5849e2ec7a0cbe573067299cf59ccf59db0cacd8db8800bb0
```
It's possible upstream re-published this version, I'm getting a checksum mismatch. 
```
  prometheus: WARN: Checksum invalid for v2.7.1.tar.gz:
   prometheus: WARN:    Expected: bfbeb434342a03b5849e2ec7a0cbe573067299cf59ccf59db0cacd8db8800bb0
   prometheus: WARN:    Computed: fe239e23379ce6d50c5577f515d7f576f1c6f9396ea47fd87c08fbf8b5e163fb
```

smacfarlane commented on prometheus/plan.sh:
```diff
@@ -38,7 +37,7 @@ do_unpack() {
 do_build() {
   pushd "$prom_pkg_dir/src/github.com/prometheus/prometheus" || exit 1
-  PREFIX="$pkg_prefix/bin" make build
+  USER="root" PREFIX="$pkg_prefix/bin" make build
```
Is this the cgo change you referenced in your PR?  Would you mind leaving a couple words in the PR what was broken and how this change fixed it?  Or if it's a cgo-ism,  are there any docs we could link to?

smacfarlane commented:
@predominant gentle ping on the above requested change

predominant commented:
Ping received. Looking at it.

predominant commented:
Fixed and rebased to include expeditor config.

The `USER=root` fix is for cgo, Its expecting the environment to have `USER` set, and in a docker studio, it is not set.
